Simulation Code/Base/Base.ino

```diff
@@ -2,46 +2,63 @@
 #include "checkWaterLevel.h"
 #include "checkPumpStatus.h"
 
-// Define the pins for the water level sensors
-const int WaterLevelA = 4;
-const int WaterLevelB = 5;
-const int WaterLevelC = 6;
-const int WaterLevelD = 7;
-
-// Define the pins for the fans
-const int FanA = 9;
-const int FanB = 10;
-int FanPWM; // Variable to control the speed of the fans
-
-// Define the variables for the water pump
-unsigned long previousTime = 0; // Variable to store the last time the water level was checked
-unsigned long previousPumpTime = 0; // Variable to store the last time the pump status was checked
-const unsigned long stabilityTime = 1000; // Time in milliseconds to wait before changing the pump speed
-bool pumpStatus = false; // Variable to store the current status of the pump (true = on, false = off)
-const unsigned long pumpTimeOff = 1000; // Time in milliseconds for the pump to stay off
-const unsigned long pumpTimeOn = 5000; // Time in milliseconds for the pump to stay on
-const int WaterPump = 11; // Pin for the water pump
-int PumpPWM; // Variable to control the speed of the water pump
+// Define the structs
+
+struct WaterLevelSensors {
+  const int WaterPinA = 4;
+  const int WaterPinB = 5;
+  const int WaterPinC = 6;
+  const int WaterPinD = 7;
+  int waterLevelA;
+  int waterLevelB;
+  int waterLevelC;
+  int waterLevelD;
+};
+
+struct Fans {
+  const int FanA = 9;
+  const int FanB = 10;
+  int FanPWM;
+};
+
+struct WaterPumpFn {
+  unsigned long previousTime = 0;
+  unsigned long previousPumpTime = 0;
+  const unsigned long stabilityTime = 1000;
+  bool pumpStatus = false;
+  const unsigned long pumpTimeOff = 1000;
+  const unsigned long pumpTimeOn = 5000;
+  const int WaterPump = 11;
+  int PumpPWM;
+  unsigned long currentTime, elapsedTime, currentPumpTime, elapsedPumpTime;
+};
+
+// Create instances of the structs
+WaterLevelSensors waterLevelSensors;
+Fans fans;
+WaterPumpFn waterPumpFn;
 
 void setup() 
 {
-  Serial.begin(9600); // Start the serial communication with the computer
-  // Set the mode of the pins
-  pinMode(WaterLevelA, INPUT);
-  pinMode(WaterLevelB, INPUT);
-  pinMode(WaterLevelC, INPUT);
-  pinMode(WaterLevelD, OUTPUT);
-  pinMode(FanA, OUTPUT);
-  pinMode(FanB, OUTPUT);
-  pinMode(WaterPump, OUTPUT);
+  Serial.begin(9600);
+
+  // Set the mode of the pins using the struct instances
+  pinMode(waterLevelSensors.WaterPinA, INPUT);
+  pinMode(waterLevelSensors.WaterPinB, INPUT);
+  pinMode(waterLevelSensors.WaterPinC, INPUT);
+  pinMode(waterLevelSensors.WaterPinD, OUTPUT);
+  pinMode(fans.FanA, OUTPUT);
+  pinMode(fans.FanB, OUTPUT);
+  pinMode(waterPumpFn.WaterPump, OUTPUT);
 }
 
 void loop() 
 {
-  int waterLevelA, waterLevelB, waterLevelC, waterLevelD; // Variables to store the current water level
-  unsigned long currentTime, elapsedTime, currentPumpTime, elapsedPumpTime; // Variables to store the current time and elapsed time
+  waterPumpFn.currentTime = millis();
```

Comment:
I'm assuming this has to be declared within the loop?

Reply:
yeah since its going to be constantly running